02_knight_and_princess/danil_dyachenko/graph_gen.cpp

```diff
@@ -0,0 +1,126 @@
+#include <array>
+#include <fstream>
+#include <iostream>
+#include <vector>
+
+using VertexId = int;
+
+struct Vertex {
+  VertexId id;
+  std::vector<int> edges;
+};
+
+struct Edge {
+  VertexId id;
+  std::array<int, 2> vertices;
+};
+
+class Graph {
+ public:
+  void add_vertices(const std::vector<Vertex>& vec) { vertices_ = vec; }
+  void print_vertices() const {
+    std::cout << "Printing vertices:" << std::endl;
+    for (int i = 0; i < vertices_.size(); ++i) {
+      std::cout << i << ": [ ";
+      for (int j = 0; j < vertices_[i].edges.size(); ++j)
+        std::cout << vertices_[i].edges[j] << " ";
+      std::cout << "]" << std::endl;
+    }
+  }
+  void add_edges(const std::vector<Edge>& vec) { edges_ = vec; }
+  void print_edges() const {
+    std::cout << "Printing edges:" << std::endl;
+    for (int i = 0; i < edges_.size(); ++i) {
+      std::cout << i << ": [ ";
+      for (int j = 0; j < edges_[i].vertices.size(); ++j)
+        std::cout << edges_[i].vertices[j] << " ";
+      std::cout << "]" << std::endl;
+    }
+  }
+  std::string to_json() {
+    std::string str;
+    str += "{\n\t\"vertices\": [\n";
+    for (int i = 0; i < vertices_.size(); ++i) {
+      str += "\t\t{ \"id\": ";
+      str += std::to_string(i);
+      str += ", \"edge_ids\": [";
+      for (int j = 0; j < vertices_[i].edges.size() - 1; ++j) {
+        str += std::to_string(vertices_[i].edges[j]);
+        str += ", ";
+      }
+      str += std::to_string(vertices_[i].edges[vertices_[i].edges.size() - 1]);
+      str += "]";
+      if (i + 1 < vertices_.size())
+        str += " },\n";
+      else
+        str += " }\n";
+    }
+    str += "\t],\n";
+    str += "\t\"edges\": [\n";
+    for (int i = 0; i < edges_.size(); ++i) {
+      str += "\t\t{ \"id\": ";
+      str += std::to_string(i);
+      str += ", \"vertex_ids\": [";
+      for (int j = 0; j < edges_[i].vertices.size() - 1; ++j) {
+        str += std::to_string(edges_[i].vertices[j]);
+        str += ", ";
+      }
+      str += std::to_string(edges_[i].vertices[edges_[i].vertices.size() - 1]);
+      str += "]";
+      if (i + 1 < edges_.size())
+        str += " },\n";
+      else
+        str += " }\n";
+    }
+    str += "\t]\n}\n";
+    return str;
+  }
+
+ private:
+  std::vector<Vertex> vertices_;
+  std::vector<Edge> edges_;
+};
+
+void write_file(const std::string& str, std::string name) {
+  std::ofstream file;
+  file.open(name, std::ios::out);
+  file << str;
+}
+
+int main(void) {
+  Graph my_graph;
+  my_graph.add_vertices((std::vector<Vertex>){{0, {0, 1, 2}},
```

Comment:
Мы пытаемся избегать `C`-синтаксиса, а это - `C-style cast`
Лучше будет:
```suggestion
const auto vertices = std::vector<Vertex>({ ... });
my_graph.add_vertices(vertices);
```